### PR TITLE
Use sparse checkout in build job

### PR DESCRIPTION
**Pull request by hallipr**
Use sparse checkout in Build job to reduce job run time

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/jobs/ci.yml [hunk 1]
```diff
@@ -52,6 +52,11 @@ jobs:
       Codeql.SkipTaskAutoInjection: false
 
     steps:
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.yml:55]: Didn't you say this break sourcelink?

Comment (hallipr) [reply eng/pipelines/templates/jobs/ci.yml:55]: Correct.   I'm closing the PR.  We can't use sparese checkouts with source link until they release a package with this fix:
https://github.com/dotnet/sourcelink/pull/772 for this issue:

Comment (hallipr) [reply eng/pipelines/templates/jobs/ci.yml:63]: Sparse checkout support was added to Source Link in version 8.0.0.  Re-opening this PR to bump sourcelink version.

**Review by weshaggard — APPROVED** [review]

**Comment by benbp** [comment]
Oh does this work now with the new sourcelink?!

**Comment by hallipr** [comment]
> Oh does this work now with the new sourcelink?!

It does.  SourceLink fixed sparse checkout support in https://github.com/dotnet/sourcelink/pull/772

**Review by jsquire — APPROVED** [review]

**Pull request merged**